src/bootstrap/compile.rs

```diff
@@ -198,6 +198,10 @@ pub fn rustc<'a>(build: &'a Build, target: &str, compiler: &Compiler<'a>) {
     if !build.unstable_features {
         cargo.env("CFG_DISABLE_UNSTABLE_FEATURES", "1");
     }
+    // Flag that rust llvm is in use
+    if build.config.target_config.get(target).is_none() {
```

Comment:
I think that this needs to check the `llvm-config` field in the returned table, as this is currently checking if there is *any* configuration at all for the target.